core/test/unit/web/middleware/url-redirects_spec.js

```diff
@@ -61,141 +62,152 @@ describe('UNIT: url redirects', function () {
             done();
         });
 
-        it('url and admin url are equal, but protocol is different, request is http', function (done) {
-            configUtils.set({
-                url: 'http://default.com:2368',
-                admin: {
-                    url: 'https://default.com:2368'
-                }
-            });
-
-            host = 'default.com:2368';
-            res.isAdmin = true;
-
-            req.originalUrl = '/ghost';
-            urlRedirects(req, res, next);
-            next.called.should.be.false();
-            res.redirect.calledWith(301, 'https://default.com:2368/ghost/').should.be.true();
-            res.set.called.should.be.true();
-            done();
-        });
-
-        it('url and admin url are different, request is http', function (done) {
-            configUtils.set({
-                url: 'http://default.com:2368',
-                admin: {
-                    url: 'https://admin.default.com:2368'
-                }
-            });
-
-            host = 'default.com:2368';
-            res.isAdmin = true;
-
-            req.originalUrl = '/ghost';
-            urlRedirects(req, res, next);
-            next.called.should.be.false();
-            res.redirect.calledWith(301, 'https://admin.default.com:2368/ghost/').should.be.true();
-            res.set.called.should.be.true();
-            done();
-        });
-
-        it('subdirectory', function (done) {
-            configUtils.set({
-                url: 'http://default.com:2368/blog',
-                admin: {
-                    url: 'https://admin.default.com:2368'
+        [
```

Comment:
Made this iterative check to avoid duplication of tests. The only "ugly" thing left is the if below, but seems better then having multiple versions of the same `.it()` :thinking:  @kirrg001 what do you think?

Reply:
Hmm yeah that is not rly nice 🤔 

Suggestion:

Unit test that these exports call `__private.redirect` with correct params.

> const adminRedirect = function adminRedirect(req, res, next) {
> const urlRedirects = function urlRedirects(req, res, next) {

Unit test `_private.redirects` separetly.

Reply:
@gargol Did you see the comment? :)